testkit/src/main/java/com/akkaserverless/javasdk/testkit/EventSourcedResult.java

```diff
@@ -26,55 +26,52 @@
  *
  * @param <R> The type of reply that is expected from invoking command handler
  */
-// FIXME a way to inspect other effects than reply?
-public final class EventSourcedResult<R> {
+public interface EventSourcedResult<R> {
 
-  private final R reply;
-  private final List<Object> events;
-  private final Iterator<Object> eventsIterator;
+  /** @return true if the call had an effect with a reply, false if not */
+  boolean isReply();
 
   /**
-   * INTERNAL API
-   *
-   * <p>Constructed by the generated code, not intended for calls from user code.
+   * The reply object from the handler if there was one. If the call had an effect without any reply
+   * an exception is thrown
+   */
+  R getReply();
+
+  /** @return true if the call was forwarded, false if not */
+  boolean isForward();
+
+  /**
+   * An object with details about the forward. If the result was not a forward an exception is
+   * thrown
+   */
+  ServiceCallDetails<R> getForward();
+
+  /** @return true if the call was an error, false if not */
+  boolean isError();
+
+  /** The error description. If the result was not an error an exception is thrown */
+  String getError();
+
+  /** @return true if the call had a noReply effect, false if not */
+  boolean isNoReply();
+
+  /**
+   * @return The updated state. If the state was not updated (no events emitted) an exeption is
+   *     thrown
    */
-  public EventSourcedResult(R reply, List<Object> events) {
-    this.reply = reply;
-    this.events = events;
-    this.eventsIterator = events.iterator();
-  }
+  Object getUpdatedState();
```

Comment:
shall we pair it with a `wasStateUpdated` to align with the ValueEntity?

Reply:
Here we have the was any events emitted method instead, I think having both does not add much.

Reply:
BTW It's a bit messy on `ValueEntityResult` anyway, since we'd have to add a noisy type param to keep the type of the state which I don't think is worth it, the testkit is typed with the state already and also has a getState of the right type which is likely more useful.

Reply:
Emitting events doesn't mean state update.

That could be a mistake, like returning the current state without updating it. Or I could have emitted events and decide to not apply. 

Actually, I think it would make more sense to have `getState` in both APIs and a check like `stateWasUpdated`. `getState` would throw exception if state was deleted in a ValueEntity